bundle/EventListener/LegacyKernelListener.php

```diff
@@ -41,8 +44,22 @@ function() {
         );
     }
 
+    public function onConfigScopeChange( ScopeChangeEvent $event )
+    {
+        $this->resetKernelHandler();
+    }
+
     public function onConsoleCommand( ConsoleCommandEvent $event )
     {
+        $this->resetKernelHandler();
+
+        $eventDispatcher = $this->container->get( 'event_dispatcher' );
```

Comment:
Please inject the event dispatcher. I know we have the container, but always prefer injection over service locator.